### PR TITLE
deskflow: renamed from synergy-core

### PR DESCRIPTION
**Pull request by sithlord48**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Aliases/synergy [hunk 1]
```diff
@@ -1 +1 @@
-../Formula/s/synergy-core.rb
\ No newline at end of file
+../Formula/d/deskflow.rb
```
Comment (nbolton) [inline Aliases/synergy:1]: Perhaps we should also have a new alias from `synergy-core` -> `deskflow`

File: formula_renames.json [hunk 2]
```diff
@@ -152,6 +152,7 @@
   "sonar": "sonarqube",
   "speedtest_cli": "speedtest-cli",
   "ssreflect": "math-comp",
+  "synergy-core":"deskflow",
```
Comment (nbolton) [inline formula_renames.json:155]: Would it also make sense to do this?

```suggestion
  "synergy":"deskflow",
  "synergy-core":"deskflow",
```

File: Aliases/synergy (hunk 1 above)
Comment (sithlord48) [reply Aliases/synergy:1]: the docs say for rename to add it to the rename json nothing about an alias no problem making one if that is what we need todo

Comment (nbolton) [reply Aliases/synergy:1]: I wonder if the `Aliases` dir is legacy? Maybe we could add it and then take it out if brew don't want it.

File: Formula/d/deskflow.rb [hunk 3]
```diff
@@ -41,36 +40,25 @@ class SynergyCore < Formula
...
-  depends_on "openssl@3"
-  depends_on "pugixml"
-  depends_on "qt"
+  depends_on cmake => :build
+  depends_on cli11
+  depends_on :macos
+  depends_on openssl@3
+  depends_on pugixml
+  depends_on qt
+  depends_on tomlplusplus
```
Comment (nbolton) [inline Formula/d/deskflow.rb:49]:
```suggestion
  depends_on "cmake" => :build
  depends_on "cli11"
  depends_on :macos
  depends_on "openssl@3"
  depends_on "pugixml"
  depends_on "qt"
  depends_on "tomlplusplus"
```

File: Formula/d/deskflow.rb [hunk 4]
```diff
@@ -1,33 +1,32 @@
...
   # This repository contains old 2.0.0 tags, one of which uses a stable tag
   # format (`v2.0.0-stable`), despite being marked as "pre-release" on GitHub.
   # The `GithubLatest` strategy is used to avoid these old tags without having
   # to worry about missing a new 2.0.0 version in the future.
   livecheck do
     url :stable
-    regex(/[^"' >]*?v?(\d+(?:\.\d+)+)[^"' >]*?/i)
+    regex(/^v\d+.\d+.\d+/i)
     strategy :github_latest
   end
```
Comment (chenrui333) [inline Formula/d/deskflow.rb:32]: need to remove the whole bottle block per formula rename

Comment (sithlord48) [reply Formula/d/deskflow.rb:32]: are you talking about the `bottle do` section ?

**Comment by sithlord48** [comment]
Is it possible to have our GUI app linked into /Applications or ~/Applications ? 

I have tried a few things in the post_install step (ln_s and osascript to tell finder to make a link ) but the step seams to always fail. 

We also have a tap where we have cask and -dev cask versions of deskflow can we have them conflict with each other i added to the recipe but it tells me i can only have one .  Is the following correct ? 

(for  the cask) 
```
conflicts_with cask: "deskflow-dev"
conflicts_with formula: "deskflow"
```
(in the formula) 

```
conflicts_with cask: [ "deskflow, "deskflow-dev" ]
```

**Comment by SMillerDev** [comment]
> Is it possible to have our GUI app linked into /Applications or ~/Applications ?

No, all formula are fully restricted to the install location of Homebrew.

**Comment by sithlord48** [comment]
> > Is it possible to have our GUI app linked into /Applications or ~/Applications ?
> 
> No, all formula are fully restricted to the install location of Homebrew.

Then should we only have a cask since we ship a GUI ?

**Comment by SMillerDev** [comment]
> Then should we only have a cask since we ship a GUI ? How do we make the GUI visible as "installed" otherwise ?

If no CLI exists anymore this should be switched to a cask. If there is a CLI, you might want to switch the install to that, and make the GUI a cask.

**Comment by sithlord48** [comment]
All of our deployments are the gui and two cli executables inside the `Deskflow.app` bundle. After building this recipe simlinks the apps inside bundle to the opt prefix.

**Comment by SMillerDev** [comment]
So Linux is no longer supported?

**Comment by sithlord48** [comment]
> So Linux is no longer supported?

it is but you can't build for any linux using wayland using brew since we require libportal and libei neither of them are on brew

**Comment by sithlord48** [comment]
Our solution has been to make our own [homebrew-tap](https://github.com/deskflow/homebrew-tap)  with the recipes to install deskflow's prebuild artifacts from github release.  

I am unsure if this recipe should be removed or just note the user should add deskflow's tap or be updated ( i would prefer the last case personally its just easier for users that way) 

The way deskflow is built the only build option for mac os is build as an app bundle. It is possible for me to add an option to not build as a bundle for macos. I would be willing todo in order to get the recipe updated here. I would need some guidance when updating

**Comment by sithlord48** [comment]
Soon this maybe possible (again) as we have added a build option to allow you to build the mac os version without bundling it up

**Pull request closed without merging**